### ISSUE TITLE
Prefix clash with MetricCollection with 1.1.1

### ISSUE THREAD

**Issue by abcamiletto** [Lightning-AI/torchmetrics#2065]
## 🐛 Bug

There is a bug that happens with metrics that returns dictionaries when used in MetricCollection.

### To Reproduce

Steps to reproduce the behavior...

```python
import torch
from torchmetrics import Metric, MetricCollection

class CustomAccuracy(Metric):
    def update(self, preds: torch.Tensor, target: torch.Tensor) -> None:
        self.correct = torch.sum(preds == target)
        self.total = preds.numel()
        self.prefix = 'accuracy'

    def compute(self) -> torch.Tensor:
        res = self.correct.float() / self.total
        return {f"{self.prefix}/value": res}

class CustomPrecision(Metric):
    def update(self, preds: torch.Tensor, target: torch.Tensor) -> None:
        self.true_positives = torch.sum((preds == target) & (preds == 1))
        self.predicted_positives = torch.sum(preds == 1)
        self.prefix = 'precision'

    def compute(self) -> torch.Tensor:
        res =  self.true_positives.float() / self.predicted_positives
        return {f"{self.prefix}/value": res}

# Initialize MetricCollection with Accuracy and Precision
metrics = MetricCollection([CustomAccuracy(), CustomPrecision()])

# Mock predictions and targets
preds = torch.tensor([1, 0, 0, 1])
targets = torch.tensor([1, 0, 0, 0])

# Update metrics with current batch
metrics(preds, targets)

# Print the calculated metrics
print(metrics.compute())
# Returns {'precisionaccuracy/value': tensor(0.7500), 'precisionprecision/value': tensor(0.5000)}
```

<details>
  <summary>Code sample</summary>

</details>

### Expected behavior

I expect the results to be 

`{'accuracy/value': tensor(0.7500), 'precision/value': tensor(0.5000)}`

and not 

`{'precisionaccuracy/value': tensor(0.7500), 'precisionprecision/value': tensor(0.5000)}`

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source): 1.1.1, via pip
- Python & PyTorch Version: 3.10 and 1.13
- Any other relevant information such as OS (e.g., Linux):

### Additional context

Please note that this works correctly in torchmetrics 0.11.4
